dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/AggregationLevelsHelper.java

```diff
@@ -0,0 +1,272 @@
+    return jdbcTemplate.query(
+        "show partitions from " + quote(tableName), (rs, rowNum) -> rs.getString("PartitionName"));
+  }
+
+  /** Fetches all column names from the target table. */
+  private List<String> getColumnNames(Table table) {
+
+    if (table.getColumns().isEmpty()) {
+      Table parent = table.getParent();
+      if (parent != null) {
+        return parent.getColumns().stream().map(Column::getName).toList();
+      }
+    } else {
+      return table.getColumns().stream().map(Column::getName).toList();
+    }
+    throw new IllegalQueryException("No columns found for table: " + table.getName());
+  }
+
+  /**
+   * Applies aggregation levels to the given analytics table, setting certain organization unit
+   * level columns (e.g., level1, level2, ...) to NULL for rows with a higher oulevel than the
+   * specified aggregation level and matching data elements.
+   *
+   * <p>The actual strategy depends on the underlying database:
+   *
+   * <ul>
+   *   <li>If {@code supportsUpdate} is true (PostgreSQL), an UPDATE statement is used.
+   *   <li>If {@code supportsUpdate} is false (Doris), a staging table and an INSERT/DELETE approach
+   *       is used.
+   * </ul>
+   *
+   * @param table the analytics table to update
+   * @param dataElements a collection of data element identifiers to match in the "dx" column
+   * @param aggregationLevel the maximum oulevel to retain; higher levels will be nullified
+   * @param supportsUpdate true for PostgreSQL (UPDATE), false for Doris (INSERT/DELETE)
+   */
+  public void applyAggregationLevels(
+      Table table, Collection<String> dataElements, int aggregationLevel, boolean supportsUpdate) {
+    if (dataElements.isEmpty()) {
+      log.warn("No data elements provided for aggregation level update.");
+      return;
+    }
+    if (supportsUpdate) {
+      withUpdate(table, dataElements, aggregationLevel);
+    } else {
+      withInsertAndDelete(table, dataElements, aggregationLevel);
+    }
+  }
+
+  /**
+   * Performs aggregation level updates using the "Insert and Delete" strategy for Apache Doris.
+   *
+   * <p>This method is used when the underlying database is Doris, which does not support multi-row
+   * UPDATE statements efficiently. It creates a staging table, inserts the updated rows (with
+   * selected level columns set to NULL) into the staging table, deletes affected rows from the main
+   * table (using partition-aware deletion if needed), and then reinserts the updated rows back into
+   * the main table. The staging table is dropped at the end.
+   *
+   * <p>
+   *
+   * @param table the Doris table to update
+   * @param dataElements a collection of data element identifiers to match in the "dx" column
+   * @param aggregationLevel the maximum oulevel to retain; higher levels will be nullified
+   */
+  void withInsertAndDelete(Table table, Collection<String> dataElements, int aggregationLevel) {
```

Comment:
I believe this helper class should be a Component, so we can make this method transactional to ensure atomic execution.